### PR TITLE
Add complex menu option

### PR DESCRIPTION
**Pull request by FlyersPh9**
- `.iui-menu` can be applied to a `<ul>` or `<ol>` to display as a basic list menu.
- `.iui-menu` can be applied to a `<div>` to display as a complex menu.
- Complex menu content must go inside `.iui-complex-content` to get the correct padding.
- If `.iui-complex-content` and `<ul>` are placed next to each other, a dividing line is added.
- Removed `.iui-menu` from `/inputs/classes/` and am instead importing `.iui-menu` from `/menu/`.
- Added tests.

Screen Shot 2021-04-21 at 2 49 00 PM
Screen Shot 2021-04-21 at 2 49 12 PM

### CODE REVIEW

**Comment by FlyersPh9** [comment]
Hey @veekeys & @bentleyvk 👋 I shared this draft PR with @mayank99 because I know it's not ready for review just yet.  However I would like your input on the overall idea.

**Comment by mayank99** [comment]
I'm bittersweet about the whole thing.

It makes sense to have a "complex" variant of menu that's generic enough to be used both in a column filter and elsewhere. It does share some styles with the basic variant (i.e.. background and shadow).

But I do not like the implementation, particularly the use of `iui-content` which can conflict with the list item, as well as the absence of list completely, which makes me wonder if it can even be considered a "menu" if it has no "menu items". So maybe we need to extract out the shared styles and reuse them to create a completely different component.

---

Slightly off topic but since the primary use case for this is column filtering, it made me curious and I did some research.

I cannot even find any accessibility guides or any reading material about the kind of filtering in our standards. Even this detailed inclusive components guide does not mention it: https://inclusive-components.design/data-tables/

Then I decided to look into some popular design systems, and here's what I found:
- Material wants you to use a [**filter row**](https://material.io/components/data-tables#anatomy) with "chips" (i.e. our `tag` component) and/or an [**search bar**](https://material.angular.io/components/table/overview#filtering:~:text=A%20general%20strategy%20is%20to%20add%20an%20input%20where%20users%20can%20type%20in%20a%20filter%20string) instead of a menu.
image. 


- [Fluent UI table](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/basic) does not have filtering built in, and uses search bar. 

- [Blueprint.js table](https://blueprintjs.com/docs/#table/features) does not even mention filtering (and it's a very mature library). 

- [Kendo UI grid](https://www.telerik.com/kendo-angular-ui/components/grid/filtering/) is the only one I found which supports the dropdown filtering flow from our standards (but it's also a library not known for modern UX). 

**TL;DR**: Maybe I'm just overthinking it, but our filtering standards could use a rework. For now, we can just provide this component according to the current standards.

**Comment by bentleyvk** [comment]
I thought making separate component with its own CSS and HTML for filters. Having this stuff in Menu CSS can cause some random bugs and can be hard to maintain. And as Mayank said it's not really a menu when there is no menu items maybe except this one:
image

**Comment by FlyersPh9** [comment]
I made some additional changes & promoted the PR from draft.  @bentleyvk do these changes help avoid the potential bugs you were talking about?

File: backstop/tests/menu.html [hunk 1]
```diff
@@ -23,22 +39,48 @@
...
-      #demo-default {
+      #demo-list {
         width: 50%;
       }
+
+      .iui-inline input {
+        width: 218px;
+      }
+
+      #demo-right-align {
```
Comment (bentleyvk) [inline backstop/tests/menu.html:50]: These rules will be needed in react and we try to avoid adding CSS rules there. Why can't we have separate CSS for filters and leaving menu as before?

**Comment by veekeys** [comment]
After reading @mayank99 research, I would suggest firstly to figure out, how our filtering should look like. @FlyersPh9 you sure we want to have those dropdowns? I like them, so it is not that I am against, but maybe we should firstly decide, whats gonna be the approach on filtering and then have implementation.
Honestly, I think current approach (reusing menu) overcomplicates things.. It is not that we need to build the filter window out of menu (it does not really make sense to me too). Even those "shared" styles are what.. background color and shadows? I do not think thats a solid ground to base filtering on menu component...
